rye/tests/test_publish.rs

```diff
@@ -0,0 +1,151 @@
+use crate::common::{fixtures_path, rye_cmd_snapshot, Space};
+
+mod common;
+
+// Test publish command
+#[test]
+fn test_publish() {
+    let space = Space::new();
+    space.init("test-project");
+
+    let fixtures_path = fixtures_path();
+    let wheel_path = fixtures_path.join("twine-1.5.0-py2.py3-none-any.whl");
+
+    let credentials_path = space.rye_home().join("credentials");
+    let credentials = std::fs::read_to_string(&credentials_path).unwrap();
+
+    // prompt for token
+    rye_cmd_snapshot!(space.rye_cmd().arg("publish").arg(&wheel_path), @r###"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    No access token found, generate one at: https://pypi.org/manage/account/token/
+
+    ----- stderr -----
+    Access token: error: IO error: not a terminal
+
+    Caused by:
+        not a terminal
+    "###);
+
+    rye_cmd_snapshot!(space.rye_cmd().arg("publish").arg("--skip-save-credentials").arg(&wheel_path), @r###"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    No access token found, generate one at: https://pypi.org/manage/account/token/
+    Uploading distributions to https://upload.pypi.org/legacy/
```

Comment:
This shouldn't happen here, but I might be misunderstanding how prompts would be handled by the tests.